### PR TITLE
Index tests in the wrong places

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/indexes/datetimes/test_astype.py [hunk 1]
```diff
@@ -117,6 +117,16 @@ def test_astype_datetime64(self):
                                  dtype='datetime64[ns]')
         tm.assert_index_equal(result, expected)
 
+    def test_astype_object(self):
+        # NumPy 1.6.1 weak ns support
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_astype.py]: can remove this comment FYI

File: pandas/tests/indexes/datetimes/test_setops.py [hunk 2]
```diff
@@ -11,14 +11,31 @@
...
+    tz = [None, 'UTC', 'Asia/Tokyo', 'US/Eastern', 'dateutil/Asia/Singapore',
+          'dateutil/US/Pacific']
 
     def test_union(self):
-        i1 = Int64Index(np.arange(0, 20, 2))
-        i2 = Int64Index(np.arange(10, 30, 2))
-        result = i1.union(i2)
-        expected = Int64Index(np.arange(0, 30, 2))
-        tm.assert_index_equal(result, expected)
+        for tz in self.tz:
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_setops.py]: don't use loops, parametrize

File: pandas/tests/indexes/datetimes/test_setops.py [hunk 3]
```diff
@@ -155,6 +172,27 @@ def test_intersection_bug_1708(self):
         result = index_1 & index_2
         assert len(result) == 0
 
+    def test_difference(self):
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_setops.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
Just pushed with parameterizing moved tests.  There are a bunch of tests elsewhere that use for loops over `self.tz`.  I'll get to them at some point in this sequence of PRs.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**